.circleci/config.yml

```diff
@@ -1,68 +1,84 @@
-post_checkout: &post_checkout
-  run:
-    name: Checkout submodule and get changed files
-    command: |
-        git submodule sync
-        git submodule update --init --recursive
-        if [[ $CIRCLE_PULL_REQUEST ]]; then
-          BASE_COMMIT=$(curl -s -H "Authorization: token a61ecb2fc5b72da54431""1b3db3875c96854958a8" \
-            https://api.github.com/repos/$CIRCLE_PROJECT_USERNAME/$CIRCLE_PROJECT_REPONAME/pulls/$CIRCLE_PR_NUMBER \
-            | jq -r ".base.sha | select(. != null)")...
-        else
-          BASE_COMMIT=HEAD^
-        fi
-        if [[ $BASE_COMMIT ]]; then
-          CHANGED=$(git diff --name-only $BASE_COMMIT)
-          echo "$CHANGED"
-          echo "export CHANGED=\"$CHANGED\"" >> $BASH_ENV
-        fi
+version: 2.1
 
-node_modules_cache_key: &node_modules_cache_key
-  yarn-deps-v2-{{ checksum "yarn.lock" }}
-
-restore_node_modules_cache: &restore_node_modules_cache
-  restore_cache:
-    keys:
-      - *node_modules_cache_key
-      - yarn-deps-v2-
+commands:
+  checkout_repo:
+    steps:
+      - checkout
+      - run:
+          name: Checkout submodule
+          command: |
+              git submodule sync
+              git submodule update --init --recursive
+      - run:
+          name: Get changed files
+          command: |
+              if [[ $CIRCLE_PULL_REQUEST ]]; then
+                BASE_COMMIT=$(curl -s -H "Authorization: token a61ecb2fc5b72da54431""1b3db3875c96854958a8" \
+                  https://api.github.com/repos/$CIRCLE_PROJECT_USERNAME/$CIRCLE_PROJECT_REPONAME/pulls/$CIRCLE_PR_NUMBER \
+                  | jq -r ".base.sha | select(. != null)")...
```

Comment:
What does `jq -r ...` do?

Reply:
@kevinbarabash It parses `json`, it is equivalent to
```js
var result = JSON.parse(pipe).base.sha;
if (result != null) return result;
return "";
```
in JS.

Reply:
Nice!